### PR TITLE
feat: extended downgrade command to include sql migrations

### LINKED ISSUE #22775: Extend `influxd downgrade` to also downgrade sqlite migrations
Extend the CLI command from #22774 to:
1. Take a `--sqlite-path` option, and
2. Run "down" migrations for all migrations marked as "up" in the sqlite DB that were added between the source & target versions of `influxd`

### PR DESCRIPTION
**Pull request by wbaker85**
Closes #22775

### CODE REVIEW

File: cmd/influxd/launcher/launcher.go [hunk 1]
```diff
@@ -986,7 +986,7 @@ func (m *Launcher) openMetaStores(ctx context.Context, opts *InfluxdOpts) (strin
 
 		// If a sqlite-path is not specified, store sqlite db in the same directory as bolt with the default filename.
 		if opts.SqLitePath == "" {
-			opts.SqLitePath = filepath.Dir(opts.BoltPath) + "/" + sqlite.DefaultFilename
+			opts.SqLitePath = filepath.Join(filepath.Dir(opts.BoltPath), sqlite.DefaultFilename)
```
Comment (wbaker85) [inline cmd/influxd/launcher/launcher.go:989]: Found this while working on the similar logic in the `downgrade` command. Seemed bad to create a path like this for cross-platform stuff.

File: cmd/influxd/downgrade/downgrade.go [hunk 2]
```diff
@@ -79,7 +90,13 @@ The target version of the downgrade must be specified, i.e. "influxd downgrade 2
 			Flag:    "bolt-path",
 			Default: filepath.Join(v2dir, bolt.DefaultFilename),
 			Desc:    "path for boltdb database",
-			Short:   'm',
+			Short:   'b',
```
Comment (danxmoran) [inline cmd/influxd/downgrade/downgrade.go:93]: `m` is the short form used for `bolt-path` in `influxd upgrade`. I thought it was also used in the main `influxd` but I seem to have imagined that... my preference would be to drop the `Short` forms of both this and `sqlite-path`

File: cmd/influxd/downgrade/downgrade.go [hunk 3]
```diff
@@ -92,10 +109,28 @@ The target version of the downgrade must be specified, i.e. "influxd downgrade 2
...
+
 	return cmd, nil
 }
 
-func downgrade(ctx context.Context, boltPath string, targetVersion string, log *zap.Logger) error {
+func downgrade(ctx context.Context, boltPath, sqlitePath, targetVersion string, log *zap.Logger) error {
+	info := influxdb.GetBuildInfo()
+
+	// Files must exist at the specified paths for the downgrade to work properly. The bolt and sqlite "open" methods will
+	// create files if they do not exist, so their existence must be verified here.
```
Comment (danxmoran) [inline cmd/influxd/downgrade/downgrade.go:123]: Nice catch

File: cmd/influxd/downgrade/downgrade.go (hunk 2 above)
Comment (wbaker85) [reply cmd/influxd/downgrade/downgrade.go]: Gotcha. Makes sense to drop the short forms here so I went ahead and did that.

**Review by danxmoran — APPROVED** [review]

**Pull request merged**